### DISCUSSION TITLE
How to integrate prisma with serverless framework aws-nodejs-typescript template

### DISCUSSION THREAD

**Discussion by kaykhan** [prisma/prisma#14259] [Q&A] [answered]
I am building a serverless function using the serverless framework. However im having an issue with running it locally

> Error: ENOENT: no such file or directory,
> open ''/.esbuild/.build/node_modules/.prisma/client/schema.prisma'

**prisma/schema.prisma**

    generator client {
        provider      = "prisma-client-js"
        binaryTargets = ["native", "rhel-openssl-1.0.x"]
    }

**serverless.ts** 

    package: {
        individually: true,
        patterns: [
            "!node_modules/.prisma/client/libquery_engine-*",
            "node_modules/.prisma/client/libquery_engine-rhel-*",
            "!node_modules/prisma/libquery_engine-*",
            "!node_modules/@prisma/engines/**",
        ],
    },

steps:

    npx prisma generate && npm install 
    
    sls invoke local -f main

What am i doing wrong here?

**note**: 

- attempting to follow this example: https://github.com/prisma/prisma-examples/tree/latest/deployment-platforms/aws-lambda

- i am using the serverless aws-nodejs-typescirpt template which uses serverless-esbuild and not serverless-webpack

**Comment by austincrim** [comment, marked as answer]
Hey there @kaykhan 👋,

Could you share the `aws-nodejs-typescript` template you are using? I was unable to reproduce this with the given info.

Ideally, you could share a minimal reproduction repository.

**Reply by kaykhan** [reply]
serverless create --template aws-nodejs-typescript --path myService

**Reply by austincrim** [reply]
I think `serverless-esbuild` is the issue here. When using `serverless-webpack`, we recommend the `serverless-webpack-prisma` plugin to copy the schema file and remove unnecessary engine binaries when packaging.

`serverless-esbuild` is building the functions but it is not configured to copy the `schema.prisma` file into the built directories, resulting in the error message you see. You could try to configure it yourself based on [`serverless-webpack-prisma`](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-aws-lambda#deployment-using-serverless-webpack).

**Reply by kaykhan** [reply]
No, the issue is that in the prisma documentation there is no mention that you need to add the schema.prisma to be included. 

The package serverless-webpack-prisma does that for you, but the problem is not serverless-esbuild and the solution serverless-webpack(-prisma). The problem is not knowing the schema.prisma needed to be bundled. 
https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-aws-lambda#package-pattern-in-serverlessyml

Once you add it to the patterns it will work

```
package: {
    individually: true,
    patterns: [
         ...
         "node_modules/.prisma/client/schema.prisma"
         ...
    ],
},

```

**Reply by janpio** [reply]
How do we go from the `create` with the template, to a project that requires the pattern added? As far as I am aware, that worked automatically in the past (but we might not have explicitly tested this template. I think we usually just test hand written projects - a hole in our coverage). Is the template by default bundling in a way that the schema file is excluded?

**Reply by kaykhan** [reply]
This is a very common template for anyone who wants to build a typescript serverless lambda function. - listed on the serverless framework docs here https://www.serverless.com/framework/docs/providers/aws/cli-reference/create and it can be bootstraped using the following command.

```
serverless create --template aws-nodejs-typescript --path app1
```

**I am not aware of the schema being automatically bundled in the past**. Since i started to use prisma a few months back, using prisma with serverless i've used the above method of adding the schema to the pattern alongside the engine. (including the engine has its own set of issues? workaround you must do to get it working locally and in deployed env). https://github.com/prisma/prisma/issues/14258

If there was one thing i'd like to ask: is for the examples repo to be updated - https://github.com/prisma/prisma-examples to include 

1. an example of using serverless aws-nodejs-typescript template, packaging the engine and schema for both local and deployed use. 

2. another example of using the same serverless aws-nodejs-typescript template with prisma layers which is what im struggling with now - https://github.com/prisma/prisma/discussions/15222.

**Reply by pantharshit00** [reply]
Hello @kaykhan 

Sorry for late reply here. I will recommend you to open an issue in `prisma-examples` repo if you want to request any new examples which you want to be included there.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
